config.go

```diff
@@ -0,0 +1,236 @@
+	file, err := ioutil.ReadFile(self.ConfigFile)
+	// don't error if the default config file isn't found
+	if os.IsNotExist(err) && self.ConfigFile == DEFAULT_CONFIG_FILE {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Could not read the config file: %s", err)
+	}
+	config := &configfile{}
+	if err = yaml.Unmarshal(file, &config); err != nil {
+		return fmt.Errorf("Could not parse the config file: %s", err)
+	}
+	self.Files = config.Files
+	self.DestHost = config.Destination.Host
+	self.DestPort = config.Destination.Port
+	self.Protocol = config.Destination.Protocol
+	self.Hostname = config.Hostname
+	self.RefreshInterval = config.RefreshInterval
+	self.ExcludeFiles = config.ExcludeFiles
+	self.ExcludePatterns = config.ExcludePatterns
+	return nil
+}
+
+func (self *Config) override() error {
+	configfile := ""
+	pflag.StringVarP(&configfile, "configfile", "c", "", "Path to config")
+	if configfile != "" {
+		self.ConfigFile = configfile
+	}
+	desthost := ""
+	pflag.StringVarP(&desthost, "dest-host", "d", "", "Destination syslog hostname or IP")
+	if desthost != "" {
+		self.DestHost = desthost
+	}
+	destport := 0
+	pflag.IntVarP(&destport, "dest-port", "p", 0, "Destination syslog port")
+	if destport != 0 {
+		self.DestPort = destport
+	}
+	foreground := true
+	if utils.CanDaemonize {
+		pflag.BoolVarP(&foreground, "no-detach", "D", false, "Don't daemonize and detach from the terminal")
+	}
+	self.Daemonize = !foreground
+	// facility
+	var s string
+	pflag.StringVarP(&s, "facility", "f", "user", "Facility")
+	facility, err := syslog.Facility(s)
+	if err != nil {
+		return fmt.Errorf("%s is not a designated facility", s)
+	}
+	self.Facility = facility
+	pflag.StringVar(&self.Hostname, "hostname", "", "Local hostname to send from")
+	pflag.StringVar(&self.PidFile, "pid-file", "", "Location of the PID file")
+	// severity
+	pflag.StringVarP(&s, "severity", "s", "notice", "Severity")
+	severity, err := syslog.Severity(s)
+	if err != nil {
+		return fmt.Errorf("Invalid severity: %s", s)
+	}
+	self.Severity = severity
+	// --strip-color
+	pflag.BoolVar(&self.UseTCP, "tcp", false, "Connect via TCP (no TLS)")
+	pflag.BoolVar(&self.UseTLS, "tls", false, "Connect via TCP with TLS")
+	pflag.BoolVar(&self.Poll, "poll", false, "Detect changes by polling instead of inotify")
+	pflag.StringVar(&s, "new-file-check-interval", "", "How often to check for new files")
+	if s != "" {
+		if err := self.RefreshInterval.Set(s); err != nil {
+			return err
+		}
+	}
+	_ = pflag.Bool("no-eventmachine-tail", false, "No action, provided for backwards compatibility")
+	_ = pflag.Bool("eventmachine-tail", false, "No action, provided for backwards compatibility")
+	logfile := ""
+	pflag.StringVar(&logfile, "debug-log-cfg", "", "the debug log file")
+	if logfile != "" {
+		self.DebugLogFile = logfile
+	}
+	pflag.StringVar(&self.LogLevels, "log", "<root>=INFO", "\"logging configuration <root>=INFO;first=TRACE\"")
+	pflag.Parse()
```

Comment:
Mea maxima culpa! I'll try to write a test for overriding config values.
On Tue, Jun 30, 2015 at 2:01 PM Leon Sodhi <notifications@github.com> wrote:

> In config.go
> <https://github.com/papertrail/remote_syslog2/pull/77#discussion_r33564883>
> :
>
> > +	pflag.BoolVar(&self.Poll, "poll", false, "Detect changes by polling instead of inotify")
> > +	pflag.StringVar(&s, "new-file-check-interval", "", "How often to check for new files")
> > +	if s != "" {
> > +		if err := self.RefreshInterval.Set(s); err != nil {
> > +			return err
> > +		}
> > +	}
> > +	_ = pflag.Bool("no-eventmachine-tail", false, "No action, provided for backwards compatibility")
> > +	_ = pflag.Bool("eventmachine-tail", false, "No action, provided for backwards compatibility")
> > +	logfile := ""
> > +	pflag.StringVar(&logfile, "debug-log-cfg", "", "the debug log file")
> > +	if logfile != "" {
> > +		self.DebugLogFile = logfile
> > +	}
> > +	pflag.StringVar(&self.LogLevels, "log", "<root>=INFO", "\"logging configuration <root>=INFO;first=TRACE\"")
> > +	pflag.Parse()
>
> Parsed values are only available after this line, so we'll need to move
> all the self.X = Y assignments so they're under it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/papertrail/remote_syslog2/pull/77/files#r33564883>.
>